platform/kube/controller.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"errors"
+	"log"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Controller is a collection of synchronized resource watchers
+type Controller struct {
+	client       *kubernetes.Clientset
+	resyncPeriod time.Duration
+	queue        Queue
+	controllers  []*cache.Controller
+
+	endpoints cache.Store
+	services  cache.Store
+}
+
+// NewController creates a new Kubernetes controller
+func NewController(
+	client *kubernetes.Clientset,
+	namespace string,
+	resyncPeriod time.Duration) *Controller {
+	out := &Controller{
+		client:       client,
+		resyncPeriod: resyncPeriod,
+		queue:        NewQueue(),
+	}
+
+	out.services = out.addWatcher(&v1.Service{},
+		func(opt v1.ListOptions) (runtime.Object, error) {
+			return client.Services(namespace).List(opt)
+		},
+		func(opt v1.ListOptions) (watch.Interface, error) {
+			return client.Services(namespace).Watch(opt)
+		},
+		out.notify,
+	)
+
+	out.endpoints = out.addWatcher(&v1.Endpoints{},
+		func(opt v1.ListOptions) (runtime.Object, error) {
+			return client.Endpoints(namespace).List(opt)
+		},
+		func(opt v1.ListOptions) (watch.Interface, error) {
+			return client.Endpoints(namespace).Watch(opt)
+		},
+		out.notify,
+	)
+
+	return out
+}
+
+func (c *Controller) notify(obj interface{}) error {
+	if !c.HasSynced() {
+		return errors.New("Waiting till full synchronization")
+	}
+	log.Printf("%#v", obj)
```

Comment:
I would like to use glog but I cannot include it together with Kubernetes without vendoring (due to a bug in `client-go` or `glog`). Any recommendation for another logging library?